File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    

    //if there is nums[j] such that if we do target - nums[i] = nums[j] then we return i, j

    seen := make(map[int]int)

    for i, num := range nums{
        complement := target - num

        if index, found := seen[complement]; found{
            return[]int{index, i}
        }

        seen[num] = i
    }

    return nil

    
}

File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        
        val seenNumbers = HashMap<Int, Int>()

        for (i in nums.indices) {

            val complement = target - nums[i]

            if (seenNumbers.containsKey(complement)) {
                return intArrayOf(seenNumbers[complement]!!, i)
            }

            seenNumbers[nums[i]] = i
        }

        return intArrayOf()

    }
}